rapids-cmake/cmake/download_with_retry.cmake

```diff
@@ -0,0 +1,94 @@
+#=============================================================================
+# Copyright (c) 2024-2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+#=============================================================================
+
+#[=======================================================================[.rst:
+rapids_download_with_retry
```

Comment:
This needs docs. Please create a new file `docs/command/rapids_cmake_download_with_retry.rst` and add it to https://github.com/rapidsai/rapids-cmake/blob/branch-25.06/docs/api.rst.

Reply:
Also added missing api.rst changes in https://github.com/rapidsai/rapids-cmake/pull/809/commits/ee941f31d8050556ca90823e90415914886e7ed9